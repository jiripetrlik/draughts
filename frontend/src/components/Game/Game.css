div.game-status {
    clear: left;
}
