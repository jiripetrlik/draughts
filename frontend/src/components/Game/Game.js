import React from 'react';
import Chessboard from "../Chessboard/Chessboard"
import Chat from "../Chat/Chat"

function Game(props) {
    return (
        <div className="Game">
          <Chessboard size="8" pieces={props.pieces} player={props.player}/>
          <Chat messages={props.messages}/>
        </div>
      );
}

export default Game
