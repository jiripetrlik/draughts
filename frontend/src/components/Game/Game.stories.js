import React from 'react'
import Game from './Game'

export default {
  component: Game,
  title: 'Game',
};

const Template = args => <Game {...args} />;

export const Default = Template.bind({});
Default.args = {
  pieces: {
    whitepieces: [[0, 0], [0, 2]],
    blackpieces: [[6, 2], [5, 3], [4, 4]],
    whitequeens: [[4, 6]],
    blackqueens: [[6, 6], [4, 4]],
    moves: [[1, 1], [1, 3]]
  },
  messages: [
    {
        Name: "You",
        Text: "Message 1"
    },
    {
        Name: "Oponent",
        Text: "Longer message"
    },
    {
        Name: "You",
        Text: "One more message"
    }
]
}
