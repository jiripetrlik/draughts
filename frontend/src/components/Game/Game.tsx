import React from 'react';
import Chessboard from "../Chessboard/Chessboard"
import Chat from "../Chat/Chat"
import "./Game.css"
import Button from '@material-ui/core/Button';
import { sendMessage } from '../../api';

class Game extends React.Component<Readonly<any>, Readonly<any>> {

  constructor(props: any) {
    super(props)
    this.leaveGame = this.leaveGame.bind(this)
  }
  
  leaveGame() {
    const command = {
      Name: "leave-game",
      Parameters: ""
    }

    sendMessage(JSON.stringify(command))
  }
  
  render() {
    let info = ""
    let myTurn = false
    if (this.props.nextMove === this.props.player) {
      myTurn = true
    }
    
    if (this.props.nextMove.includes("won")) {
      const victor = this.props.nextMove.includes("white") ? "white" : "black"
      if (this.props.player === victor) {
        info = "You've won"
      } else {
        info = "You've lost"
      }
    } else if (this.props.nextMove === "oponent-left") {
      info = "Your oponent has left"
    } else {
      if (myTurn) {
        info = "Your turn"
      } else {
        info = "Oponent's turn"
      }
    }

    return (
      <div className="game">
        <Chessboard size="8" pieces={this.props.pieces} player={this.props.player} myTurn={myTurn}/>
        <Chat messages={this.props.messages}/>
        <Button onClick={this.leaveGame}>Leave Game</Button>
        <div className="game-status">{info}</div>
      </div>
    );
  }
}

export default Game
