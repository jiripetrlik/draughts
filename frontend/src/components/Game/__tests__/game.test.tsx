import React from 'react';
import renderer from 'react-test-renderer';
import Game from '../Game'

test('snapshot test', () => {
    const pieces = {
        whitepieces: [[0, 0], [0, 2]],
        blackpieces: [[6, 2], [5, 3], [4, 4]],
        whitequeens: [[4, 6]],
        blackqueens: [[6, 6], [4, 4]],
        moves: [[1, 1], [1, 3]]
    };
    const messages = [
        {
            name: "You",
            text: "Message 1"
        },
        {
            name: "Oponent",
            text: "Longer message"
        },
        {
            name: "You",
            text: "One more message"
        }
    ];

    const tree = renderer
        .create(<Game pieces={pieces} messages={messages} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
