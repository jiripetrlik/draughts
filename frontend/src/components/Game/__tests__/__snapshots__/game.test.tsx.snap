// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="Game"
>
  <table
    className="chessboard"
  >
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
      </tr>
      <tr>
        <td>
          2
        </td>
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
      </tr>
      <tr>
        <td>
          3
        </td>
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
      </tr>
      <tr>
        <td>
          4
        </td>
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
      </tr>
      <tr>
        <td>
          5
        </td>
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
      </tr>
      <tr>
        <td>
          6
        </td>
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
      </tr>
      <tr>
        <td>
          7
        </td>
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
      </tr>
      <tr>
        <td>
          8
        </td>
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
        <td
          className="dark"
          onClick={[Function]}
        />
        <td
          className="light"
          onClick={[Function]}
        />
      </tr>
      <tr>
        <td />
        <td>
          A
        </td>
        <td>
          B
        </td>
        <td>
          C
        </td>
        <td>
          D
        </td>
        <td>
          E
        </td>
        <td>
          F
        </td>
        <td>
          G
        </td>
        <td>
          H
        </td>
      </tr>
    </tbody>
  </table>
  <div
    id="chat"
  >
    <div>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        : 
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        : 
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        : 
      </p>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root"
      onKeyDown={[Function]}
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        Message
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
      </div>
    </div>
  </div>
</div>
`;
